aicsimageio/writers/ome_tiff_writer.py

```diff
@@ -87,31 +88,63 @@ def save(self, data, ome_xml=None, channel_names=None, image_name="IMAGE0", pixe
         if self.silent_pass:
             return
 
-        tif = tifffile.TiffWriter(self.file_path, bigtiff=self._size_of_ndarray(data=data) > BYTE_BOUNDARY)
-
         shape = data.shape
-        assert (len(shape) == 5 or len(shape) == 4 or len(shape) == 3)
-
-        # if this is 3d data, then assume it's ZYX and transform it to the expected TZCYX
-        if len(shape) == 3:
-            data = np.expand_dims(data, axis=1)
+        ndims = len(shape)
+
+        assert (ndims == 5 or ndims == 4 or ndims == 3), "Expected 3, 4, or 5 dimensions in data array"
+
+        # assert valid characters in dimension_order
+        assert (all(d in "STCZYX" for d in dimension_order)), f"Invalid dimension_order {dimension_order}"
+        assert (dimension_order[-2:] == "YX"), \
+            f"Last two characters of dimension_order {dimension_order} expected to be YX.  Please transpose your data."
+        assert (len(dimension_order) >= ndims), \
+            f"dimension_order {dimension_order} must have at least as many dimensions as data shape {shape}"
+        assert (dimension_order.find("S") == 0 or dimension_order.find("S") == -1), \
```

Comment:
nit: Wrap these in more explicit Exceptions. `InvalidDimensionOrder` or similar?